sw/device/silicon_creator/lib/drivers/alert.c

```diff
@@ -0,0 +1,175 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/alert.h"
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#include "alert_handler_regs.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+enum {
+  kBase = TOP_EARLGREY_ALERT_HANDLER_BASE_ADDR,
+};
+
+rom_error_t alert_configure(size_t index, alert_class_t cls,
+                            alert_enable_t enabled) {
+  if (index >= ALERT_HANDLER_ALERT_CLASS_MULTIREG_COUNT) {
+    return kErrorAlertBadIndex;
+  }
+  index *= 4;
+
+  switch (cls) {
+    case kAlertClassA:
+      abs_mmio_write32(kBase + ALERT_HANDLER_ALERT_CLASS_0_REG_OFFSET + index,
+                       0);
```

Comment:
nit: it looks like we have generated defines for class values, e.g. `ALERT_HANDLER_ALERT_CLASS_0_CLASS_A_0_VALUE_CLASSA`.

Reply:
Fixed.